include/qore/QoreNamespace.h

```diff
@@ -309,4 +358,130 @@ class QoreNamespaceConstIterator {
     DLLEXPORT const QoreNamespace* get() const;
 };
 
+//! allows local namespaces to be iterated
+/** @since %Qore 0.8.13
+ */
+class QoreNamespaceNamespaceIterator {
+public:
+    //! creates the iterator
+    DLLEXPORT QoreNamespaceNamespaceIterator(const QoreNamespace* ns);
+
+    //! destroys the iterator
+    DLLEXPORT ~QoreNamespaceNamespaceIterator();
```

Comment:
I like the rule where destructor is virtual, or class is final, so no one can inherit a class and take a pointer to its ancestor and then the destructor is not called, what do you think?